File: Examples/ManualExample/ManualExample/ViewController.m
//
//  ViewController.m
//  ManualExample
//
//  Created by Tony Xiao on 6/30/16.
//  Copyright © 2016 Segment. All rights reserved.
//

#import <Analytics/Analytics.h>
#import "ViewController.h"


@interface ViewController ()

@end


@implementation ViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    [[SEGAnalytics sharedAnalytics] track:@"Manual Example Main View Loaded"];
    [[SEGAnalytics sharedAnalytics] flush];
}

- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)fireEvent:(id)sender
{
    [[SEGAnalytics sharedAnalytics] track:@"Manual Example Fire Event"];
    [[SEGAnalytics sharedAnalytics] flush];
}

@end

File: Examples/SegmentMac/SegmentMac/ViewController.swift
//
//  ViewController.swift
//  SegmentMac
//
//  Created by Cody Garvin on 7/6/20.
//  Copyright © 2020 Cody Garvin. All rights reserved.
//

import Cocoa
import Segment

class ViewController: NSViewController {
    
    @IBOutlet weak var signinButton: NSButton!
    @IBOutlet weak var associateButton: NSButton!
    @IBOutlet weak var signoutButton: NSButton!

    override func viewDidLoad() {
        super.viewDidLoad()
        
        let configuration = AnalyticsConfiguration(writeKey: "GM88xxNqA6PfthZE9ZzA2AREQBMjc6xA")

        // Enable this to record certain application events automatically!
        configuration.trackApplicationLifecycleEvents = true
        // Enable this to record screen views automatically!
        configuration.recordScreenViews = true
        configuration.flushAt = 1
        Analytics.setup(with: configuration)


        // Do any additional setup after loading the view.
        let analytics = Analytics.shared()
        
        // identify screen load
        analytics.screen("SegmentMac")
    }

    override var representedObject: Any? {
        didSet {
        // Update the view, if already loaded.
        }
    }

    @IBAction func signinTapped(_ sender: Any) {
        associateButton.isEnabled = true
        Analytics.shared().track("Signed In")
    }
    
    @IBAction func associateTapped(_ sender: Any) {
        signoutButton.isEnabled = true
        Analytics.shared().alias("New-Associate-ID")
    }
    
    @IBAction func signoutTapped(_ sender: Any) {
        Analytics.shared().track("Signed Out")
    }
}

